### PR TITLE
[FRONT-70] Update sidebar to match current design

### PR DESCRIPTION
**Pull request by mohammad-akkad**
### Description of Issue
Updated loading page and side bar according to figma

### How I implemented
- removed the sliding bar and instead add static text
image
-changed the side bar according to figma
image

- Added routing with parameters and highlighting the open page
- Added hover effect with buttons
- fixed styling

### Other

### CODE REVIEW

**Review by DamianGrosz2 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "next-urql": "^5.0.1",
     "postcss": "8.4.23",
     "react": "18.2.0",
+    "react-beautiful-dnd": "^13.1.1",
```
Comment (cnk1dev) [inline package.json:34]: Remove `react-beautiful-dnd` if it's not being used yet

File: src/components/pure/Sidebar/Section/index.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ const Section = ({ children, title }: SectionProps) => (
     key="div-2"
     className="border-t border-solid border-gray-800 py-2.5 text-14 font-normal leading-4 text-gray-100"
   >
-    <h1 className="px-7 py-2.5 ">{title}</h1>
+    <h1 className="px-7 py-2.5 text-12 font-medium uppercase text-gray-400">{title}</h1>
```
Comment (cnk1dev) [inline src/components/pure/Sidebar/Section/index.tsx:11]: Use h3 or p here + make section title smaller like in Figma

File: src/components/side-effects/NavSidebar.tsx [hunk 3]
```diff
@@ -1,68 +1,147 @@
...
+        text: 'vercel / next.js',
+        onClick: nullFunc,
+        showIcon: true,
+        secondaryItem: true,
+        editable: true
+      },
+      {
+        id: 6,
+        Icon: Dot,
+        text: 'microsfot / playwright',
```
Comment (cnk1dev) [inline src/components/side-effects/NavSidebar.tsx:75]:
```suggestion
        text: 'microsoft / playwright',
```

File: src/components/pure/Sidebar/Section/Item.tsx
Comment (cnk1dev) [inline src/components/pure/Sidebar/Section/Item.tsx:1]: - [ ] Change icon for project to filled circle like in design
- [ ] Fix glitch on hover when title is too long
- [ ] Disable editing & removing actions for now (keep the icons but remove click handlers)
- [ ] Link projects to /details/<id>
- [ ] Link folders to /compare/<category>
- [ ] Fix cut off text at the bottom:
Screenshot 2023-05-29 at 15 45 35

**Review by cnk1dev — CHANGES_REQUESTED** [review]
Please have a look at my comments below. Also:
- [x] Make TruffleAI logo/ sidebar title bigger & bolder
- [ ] Add link to / on logo, trending projects & all bookmarks

File: src/components/side-effects/NavSidebar.tsx [hunk 4]
```diff
@@ -1,68 +1,147 @@
...
+        ...section,
+        items: section.items.filter((item) => item.id !== id)
+      }))
+    )
+  }
+
+  return (
+    <Sidebar title="TruffleAI" footer={renderFooter(router)}>
+      {sections.map((section) => (
+        <Sidebar.Section title={section.title}>
```
Comment (cnk1dev) [inline src/components/side-effects/NavSidebar.tsx:125]: Add key to section

File: package.json (hunk 1 above)
Comment (DamianGrosz2) [reply package.json:34]: done

File: src/components/side-effects/NavSidebar.tsx (hunk 4 above)
Comment (DamianGrosz2) [reply src/components/side-effects/NavSidebar.tsx:125]: done

File: src/components/pure/Sidebar/Section/index.tsx (hunk 2 above)
Comment (DamianGrosz2) [reply src/components/pure/Sidebar/Section/index.tsx:11]: done

File: src/components/pure/Sidebar/Section/Item.tsx
Comment (DamianGrosz2) [reply src/components/pure/Sidebar/Section/Item.tsx:1]: done

File: package.json [hunk 5]
```diff
@@ -45,6 +45,7 @@
     "@graphql-codegen/typescript": "^3.0.4",
     "@graphql-codegen/typescript-operations": "^3.0.4",
     "@graphql-codegen/typescript-urql": "^3.7.3",
+    "@types/react-beautiful-dnd": "^13.1.4",
```
Comment (pczern) [inline package.json:48]: wrong location

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/components/side-effects/NavSidebar.tsx [hunk 6]
```diff
@@ -1,68 +1,160 @@
...
+  const [sections] = useState(initialSections)
+
+  const handleSave = (id: number, newText: string) => {
+    // setSections(
+    //   sections.map((section) => ({
+    //     ...section,
+    //     items: section.items.map((item) => (item.id === id ? { ...item, text: newText } : item))
+    //   }))
+    // )
+    alert(`Saved ${newText}`)
```
[the remark this answers is not in the document]
Comment (DamianGrosz2) [reply src/components/side-effects/NavSidebar.tsx:122]: why, this is just for demo purposes

File: src/components/side-effects/NavSidebar.tsx [hunk 7]
```diff
@@ -1,68 +1,160 @@
...
+  }
+
+  const handleDelete = (id: number) => {
+    // setSections(
```
[the remark this answers is not in the document]
Comment (DamianGrosz2) [reply src/components/side-effects/NavSidebar.tsx:126]: but the functionality would work like this, once the data is there

File: src/components/side-effects/NavSidebar.tsx [hunk 8]
```diff
@@ -1,68 +1,160 @@
...
+          showIcon: true,
+          editable: true
+        }
+      ]
+    }
+  ]
+  const [sections] = useState(initialSections)
+
+  const handleSave = (id: number, newText: string) => {
+    // setSections(
```
[the remark this answers is not in the document]
Comment (DamianGrosz2) [reply src/components/side-effects/NavSidebar.tsx:116]: but the functionality would work like this, once the data is there

File: src/components/pure/Sidebar/Section/Item.tsx [hunk 9]
```diff
@@ -1,23 +1,106 @@
...
+    setIsEditable(true)
+  }
+
+  const handleChange = (e: { target: { value: SetStateAction<string> } }) => {
+    setValue(e.target.value)
+  }
+
+  const handleBlur = () => {
+    setIsEditable(false)
+    // TODO
```
[the remark this answers is not in the document]
Comment (DamianGrosz2) [reply src/components/pure/Sidebar/Section/Item.tsx]: fixed

File: package.json (hunk 5 above)
Comment (DamianGrosz2) [reply package.json]: fixed

File: src/components/side-effects/NavSidebar.tsx [hunk 10]
```diff
@@ -1,68 +1,160 @@
...
-        Icon={Bookmark}
-        text="Machine Learning"
-        onClick={nullFunc}
-        showIcon={false}
-      />
-    </Sidebar.Section>
-  </Sidebar>
-)
+const NavSidebar = ({ router }: { router: NextRouter }) => {
+  const initialSections = [
```
[the remark this answers is not in the document]
Comment (DamianGrosz2) [reply src/components/side-effects/NavSidebar.tsx:25]: done

File: src/components/side-effects/NavSidebar.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/components/side-effects/NavSidebar.tsx]: removed

File: src/components/side-effects/NavSidebar.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/components/side-effects/NavSidebar.tsx]: removed

**Review by cnk1dev — APPROVED** [review]

**Review by pczern — CHANGES_REQUESTED** [review]

File: package-lock.json
[the remark this answers is not in the document]
Comment (cnk1dev) [reply package-lock.json:1]: regenerated `package-lock.json`

**Review by pczern — APPROVED** [review]

**Pull request merged**